### PR TITLE
port freespace plannet to ros2

### PR DESCRIPTION
**Pull request by TakaHoribe**
depending on the `astar_search` package. https://github.com/tier4/Pilot.Auto/pull/38

Remarks:
 - Since `tf2::doTransform(Pose, Pose, Transform)` is not in the ROS2 tf_geometry_msgs, it is replaced to `tf2::doTransform(PoseStamped, PoseStamped, Transform)`

### CODE REVIEW

**Review by jilaada — COMMENTED** [review]
Just a couple of minor code comments similar to that of the `astar_search`. Haven't been able to test yet due to the astar dependency.

**Comment by jilaada** [comment]
I tried to compile this after merging the astar_search changes however it fails as it is unable to find the astar_search header. 

```
jilada@ade:~/AutowareArchitectureProposal (ros2-porting/add-ublox-deps $% u=)$ colcon build --packages-select astar_search                                                                                         
Starting >>> astar_search                                                                                                                                                                                                           
Finished <<< astar_search [0.21s]                                                                                                                                                                                  
                                                                                                                                                                                                                   
Summary: 1 package finished [0.43s]                                                                                                                                                                                
jilada@ade:~/AutowareArchitectureProposal (ros2-porting/add-ublox-deps $% u=)$ colcon build --packages-select freespace_planner                                                                                    
Starting >>> freespace_planner                                                                                                                                                                                     
                      
--- stderr: freespace_planner                                                                                                                                                                                      
In file included from /home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/planning/scenario_planning/parking/freespace_planner/src/freespace_planner/main.cpp:33:                                   
/home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/planning/scenario_planning/parking/freespace_planner/include/freespace_planner/freespace_planner.h:42:10: fatal error: astar_search/astar_search
.h: No such file or directory                                                                                                                                                                                      
   42 | #include <astar_search/astar_search.h>                                                                                                                                                                     
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                     
compilation terminated.                                                                                                                                                                                            
In file included from /home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/planning/scenario_planning/parking/freespace_planner/src/freespace_planner/freespace_planner_node.cpp:33:                 
/home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/planning/scenario_planning/parking/freespace_planner/include/freespace_planner/freespace_planner.h:42:10: fatal error: astar_search/astar_search
.h: No such file or directory                                                                                                                                                                                      
   42 | #include <astar_search/astar_search.h>                                                                                                                                                                     
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                     
compilation terminated.                                                                                                                                                                                            
make[2]: *** [CMakeFiles/freespace_planner.dir/build.make:76: CMakeFiles/freespace_planner.dir/src/freespace_planner/main.cpp.o] Error 1                                                                           
make[2]: *** Waiting for unfinished jobs....                                                                                                                                                                       
make[2]: *** [CMakeFiles/freespace_planner.dir/build.make:63: CMakeFiles/freespace_planner.dir/src/freespace_planner/freespace_planner_node.cpp.o] Error 1                                                         
make[1]: *** [CMakeFiles/Makefile2:78: CMakeFiles/freespace_planner.dir/all] Error 2                                                                                                                               
make: *** [Makefile:141: all] Error 2                                                                                                                                                                              
---                                                                                                                                                                                                                
Failed   <<< freespace_planner [0.15s, exited with code 2]                                                                                                                                                         
                                                                                                                                                                                                                   
Summary: 0 packages finished [0.38s]                                                                                                                                                                               
  1 package failed: freespace_planner                                                                                                                                                                              
  1 package had stderr output: freespace_planner
```

**Review by jilaada — APPROVED** [review]
Tentatively approving bar the changes I specified earlier are addressed. I was able to pull in the `astar_search` and compile the `freespace_planner` and run the node.

File: planning/scenario_planning/parking/freespace_planner/package.xml
Comment (TakaHoribe) [reply planning/scenario_planning/parking/freespace_planner/package.xml:13]: Thanks! I've changed.

File: planning/scenario_planning/parking/freespace_planner/src/freespace_planner/freespace_planner_node.cpp
Comment (TakaHoribe) [reply planning/scenario_planning/parking/freespace_planner/src/freespace_planner/freespace_planner_node.cpp]: Changed.

**Comment by TakaHoribe** [comment]
Merge this PR after https://github.com/tier4/Pilot.Auto/pull/38 is merged and then CI passed.

**Review by jilaada — APPROVED** [review]
Changes look good to me :+1:

**Pull request merged**
